tests/test_ocs_basic_install.py

```diff
@@ -34,19 +36,28 @@ def run(**kwargs):
     log.info("Generating install-config")
     # TODO: determine better place to create cluster directories - (log dir?)
     cluster_dir_parent = "/tmp"
-    base_name = test_data.get('cluster-name', 'ocs-ci-cluster')
+    cluster_name = test_data.get('cluster-name')
```

Comment:
Should we be requiring the user to pass both the cluster-name as well as the directory? If we have the directory we can infer the name based on that. I think this whole block would be cleaner if we started with a check to see if the `cluster_path` was provided. If so we set the name in the test data as well, test access to the cluster, and return SKIPPED. Otherwise, if no `cluster_path` is set, we can just do what we were doing before with the addition of setting the default value to `default.CLUSTER_NAME`.

Reply:
@clacroix12 the purpose of this was mainly cause of automation runs on Jenkins slaves. I would like to have some directory specified like /home/jenkins/cluster-dir  - as you can see I don't have any name of the cluster in the name of directory.  The jenkins node can during it's deployment create a cluster directory in the share mount point, so /home/jenkins/cluster-dir will be on all the slaves all the time the same (You always know where to find it), and we can create //nfs_share:/clusters_data/unique_cluster_name_id/ which will be mounted in /home/jenkins/cluster-dir . With solution with just cluster_path and taking the name from this directory.

Reply:
@vasukulkarni @RazTamir @dahorak  can you please take a look to this conversation and agree? So we can merge and move on or just decide to do it diff way and I can change the logic. But as I said as we will have one jenkins node per cluster, and till cluster is up we will have jenkins slave up, I prefer to have the data about the cluster in for example mentioned folder /home/jenkins/cluster-dir

Reply:
From my POV, this file 'test_ocs_basic_install.py' should be used only for deployment purposes and not part of the testing phase, hence, the user shouldn't have an already existing cluster/path/directory.
When running tests locally, we can pass the path to the AWS directory but it seems like not a convenient way

Reply:
Maybe I got the whole thing you meant @clacroix12 Is this what you've meant: https://github.com/red-hat-storage/ocs-ci/pull/27/commits/8b84809100a0211ff5e08a5f919a3036098b396b move the check at the start?

Reply:
From my POV, I agree with the flexibility provided by the suggested changes... but my experience with this framework is still very limited, so maybe I'm missing some crucial context.

Reply:
@petr-balogh yes that's essentially what I meant. However since you have the `cluster_path` and `OCP.set_kubeconfig` paired up in that `if` we will only return skipped if the cluster is ready. In the even that the cluster_path is set and in a bad state, we will continue with a deployment. Is that intended?

Reply:
@RazTamir I'm not sure I agree. The installation of OCP is something we will be testing as well, is it not? I don't see why this shouldn't be treated as a test.

Reply:
@clacroix12 actually it's just checking if OCP is up and running and you are able to connect with credentials you have in cluster_path.  I am not checking any other state of the OCS cluster and assume that user know that cluster is up and running. (We can improve in next PR to check also OCS cluster and then decide what to do if it's not ready) But as I said, I want to be able to have cluster_path specified for jenkins slave in for example /home/jenkins/ocs_cluster_dir and even in first run when cluster doesn't exist, it will be created from scratch. It's up to user to have cluster in good shape to run on when decided to run on existing cluster.

Reply:
@clacroix12 as there are another PRs kind of depend on this change or can reuse some parts from it, I would go and merge this one if you and also @vasukulkarni approve. You can merge and align those kind of depending PRs.  I can do another requests if there are some in new PR, WDYT?

Reply:
Right, I guess what I mean is if they specify a cluster-path for a cluster that isn't running, we will go forward with the installation and most likely hit other issues because we're using the same directory as another install. I think we can merge this but we should really revisit this logic here and make sure we aren't going to hit any weird cases like that.